context/src/otelInBraveTest/java/io/opentelemetry/context/BraveContextStorageProvider.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.context;
+
+import brave.Tracing;
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.TraceContext;
+import java.util.List;
+import javax.annotation.Nullable;
+
+public class BraveContextStorageProvider implements ContextStorageProvider {
+
+  @Override
+  public ContextStorage get() {
+    return BraveContextStorage.INSTANCE;
+  }
+
+  @SuppressWarnings("ReferenceEquality")
+  private enum BraveContextStorage implements ContextStorage {
+    INSTANCE;
+
+    @Override
+    public Scope attach(Context toAttach) {
+      CurrentTraceContext currentTraceContext = Tracing.current().currentTraceContext();
+      TraceContext currentBraveContext = currentTraceContext.get();
+      ContextWrapper currentContext = fromBraveContext(currentBraveContext);
+      if (currentContext == ContextWrapper.ROOT
+          && (toAttach == null || toAttach == Context.root())) {
+        // It may be possible that in the current brave we have null, and asked to add root,
+        // but the behavior of Current is to never return null, so it is fine to return noop.
+        return Scope.noop();
+      }
+
+      TraceContext newBraveContext =
+          currentBraveContext.toBuilder().addExtra(new ContextWrapper(toAttach)).build();
+      CurrentTraceContext.Scope braveScope = currentTraceContext.newScope(newBraveContext);
+      return braveScope::close;
+    }
+
+    @Override
+    public Context current() {
+      TraceContext currentBraveContext = Tracing.current().currentTraceContext().get();
+      if (currentBraveContext == null) {
+        return Context.root();
+      }
+      return fromBraveContext(currentBraveContext);
+    }
+
+    private static ContextWrapper fromBraveContext(TraceContext braveContext) {
+      ContextWrapper currentContext = findContextWrapper(braveContext.extra());
+      return currentContext == null ? ContextWrapper.ROOT : currentContext;
+    }
+
+    static ContextWrapper findContextWrapper(List<Object> extra) {
+      for (int i = extra.size() - 1; i >= 0; i--) {
+        Object nextExtra = extra.get(i);
+        if (nextExtra.getClass() == ContextWrapper.class) return (ContextWrapper) nextExtra;
+      }
+      return null;
+    }
+  }
+
+  // Need to wrap the Context because brave findExtra searches for perfect match of the class.
+  static final class ContextWrapper implements Context {
```

Comment:
Instead of this, I may be able to check for `DefaultContext` but that may not work for everyone, so to show how to do it I wrapped it.